wgengine/magicsock/magicsock.go

```diff
@@ -636,6 +648,18 @@ func (c *Conn) setEndpoints(endpoints []tailcfg.Endpoint) (changed bool) {
 	return true
 }
 
+// SetStaticEndpoints sets static endpoints to the provided value and triggers
+// an asynchronous update of the endpoints that this node advertizes.
+func (c *Conn) SetStaticEndpoints(ep []netip.AddrPort) {
+	c.mu.Lock()
+	c.staticEndpoints = ep
+	c.mu.Unlock()
+	// Technically this is not a reSTUNning, but ReSTUN does what we need at
+	// this point- calls updateEndpoints or queues an update if there is
+	// already an in-progress update.
+	c.ReSTUN("static-endpoint-change")
```

Comment:
Alternatively I could split out the updating of static endpoints into a separate method. But, assuming that even if static endpoints are set, we wil always also want the dynamic ones, it seems safer and less confusing if the bits that [set the last endpoint update etc](https://github.com/tailscale/tailscale/blob/v1.70.0/wgengine/magicsock/magicsock.go#L626-L636) are only set from one place.